### PR TITLE
V0_10 merge 0.9.41 snapshot

### PR DESCRIPTION
**Pull request by TrsNium**
To include v0.9.41-snapshot modification to v0_10, this PR merge v0.9.41-snapshot to v0_10.
v0.10 hasn't followed master recently. I think this will be a problem in adding new features

### CODE REVIEW

File: digdag-server/src/test/java/io/digdag/server/AuthRequestFilterTest.java [hunk 1]
```diff
@@ -109,25 +111,30 @@ private void verifyThatMethodIsAuthenticated(String method)
...
+        Config userInfo = CONFIG_FACTORY.create();
+        Supplier<Map<String, String>> secrets = () -> ImmutableMap.of("secret", "value");
+        Authenticator.Result acceptance = Authenticator.Result.builder()
                 .siteId(17)
-                .isAdmin(true)
                 .userInfo(userInfo)
-                .userContext(userContext)
+                .secrets(secrets)
+                .isAdmin(true)
+                .authenticatedUser(AuthenticatedUser.builder()
```
Comment (TrsNium) [inline digdag-server/src/test/java/io/digdag/server/AuthRequestFilterTest.java]: Add `authenticatedUser` to` builder` for compatibility

File: digdag-spi/src/main/java/io/digdag/spi/Authenticator.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        static Builder builder() {
+            return ImmutableResult.builder();
+        }
+
+        interface Builder {
+            Builder siteId(int siteId);
+            Builder userInfo(Config userInfo);
+            Builder secrets(Supplier<Map<String, String>> secrets);
+            Builder isAdmin(boolean admin);
+            Builder authenticatedUser(AuthenticatedUser authenticatedUser);
```
Comment (TrsNium) [inline digdag-spi/src/main/java/io/digdag/spi/Authenticator.java]: Add `authenticatedUser` to` builder` for compatibility

File: digdag-standards/src/main/java/io/digdag/standards/auth/jwt/JwtAuthenticator.java [hunk 3]
```diff
@@ -118,18 +102,15 @@ public Key resolveSigningKey(JwsHeader header, String plaintext)
...
                 .siteId(siteId)
                 .isAdmin(admin)
-                .userInfo(cf.create())
-                .userContext(cf.create())
+                .authenticatedUser(AuthenticatedUser.builder()
+                      .siteId(siteId)
+                      .isAdmin(admin)
+                      .userInfo(cf.create())
+                      .userContext(cf.create())
+                      .build())
```
Comment (TrsNium) [inline digdag-standards/src/main/java/io/digdag/standards/auth/jwt/JwtAuthenticator.java]: Add `authenticatedUser` to` builder` for compatibility

**Comment by TrsNium** [comment]
@yoyama @muga
`digdag-tests/src/test/java/acceptance/AccessController.java` could not be passed. If you realise why not this test has not been passed, please teach me them.

File: digdag-cli/src/main/java/io/digdag/cli/Server.java [hunk 4]
```diff
@@ -134,7 +135,7 @@ private void startServer()
         Properties props = buildServerProperties();
         ConfigElement ce = PropertyUtils.toConfigElement(props);
         ServerConfig serverConfig = ServerConfig.convertFrom(ce);
-        ServerBootstrap.start(buildServerBootstrap(version, serverConfig));
+        ServerBootstrap.start(buildServerBootstrap(version, serverConfig, loadSystemPlugins(props)));
```
Comment (TrsNium) [inline digdag-cli/src/main/java/io/digdag/cli/Server.java]: In master branch, initialize ServerBootstrap class directly.
But use buildServerBootstrap method to override from `digdag-tests/src/test/java/acceptance/AccessControllerIT.java`

File: digdag-tests/src/test/java/acceptance/AccessControllerIT.java [hunk 5]
```diff
@@ -138,6 +143,14 @@ public void bindAuthorization()
...
                             binder().bind(AccessController.class).to(ProjectBasedAccessController.class);
                         }
+
+                        @Override
+                        protected void bindAuthenticator()
+                        {
+                            binder().bind(JwtAuthenticatorConfig.class).toProvider(JwtAuthenticatorConfigProvider.class).asEagerSingleton();
+                            binder().bind(Authenticator.class).to(JwtAuthenticator.class);
+                        }
+
```
Comment (TrsNium) [inline digdag-tests/src/test/java/acceptance/AccessControllerIT.java]: It is necessary to overwrite `bindAuthenticator` by changing the specifications of` ServerBootstrap`.
Originally, `v0.10` uses` JwtAuthenticator` as `Authenticator`, so it binds JwtAuthenticatorConfig etc.

File: digdag-server/src/main/java/io/digdag/server/rs/AuthenticatedResource.java [hunk 6]
```diff
@@ -18,12 +18,12 @@ protected int getSiteId()
     {
         // siteId is set by JwtAuthInterceptor
         // TODO validate before causing NPE. Improve guice-rs to call @PostConstruct
-        return getAuthenticatedUser().getSiteId();
+        return (int) request.getAttribute("siteId");
```
Comment (TrsNium) [inline digdag-server/src/main/java/io/digdag/server/rs/AuthenticatedResource.java]: `master` did not use` getAuthenticatedUser`, but instead fetched the value from `HttpServletRequest`.

File: digdag-server/src/main/java/io/digdag/server/rs/AuthenticatedResource.java [hunk 7]
```diff
@@ -18,12 +18,12 @@ protected int getSiteId()
...
     }
 
     protected Config getUserInfo()
     {
-        return getAuthenticatedUser().getUserInfo();
+        return (Config) request.getAttribute("userInfo");
```
Comment (TrsNium) [inline digdag-server/src/main/java/io/digdag/server/rs/AuthenticatedResource.java]: `master` did not use` getAuthenticatedUser`, but instead fetched the value from `HttpServletRequest`.

**Comment by TrsNium** [comment]
@yoyama 
I hope you to rebase v0.10 as soon as possible

**Comment by frsyuki** [comment]
Merging master to v0_10 a MUST DO item. Maintaining two branches without being able to merge is a traditional anti-pattern.
We appreciate the work. But please let us control the risk of deployments to our production system.
@TrsNium You may want to join Treasure Data to be a part of decision maker.

**Pull request closed without merging**